### DISCUSSION TITLE
Sandpack & Multiple Instances of Preview Manager

### DISCUSSION THREAD

**Discussion by hzhu** [codesandbox/codesandbox-client#4631] [General] [answered]
Hi 👋 , I'm trying to render multiple instances of Smooshpack's [`PreviewManager`](https://github.com/codesandbox/codesandbox-client/blob/master/standalone-packages/sandpack/src/manager/index.ts#L88) on a single page.

### Problem

When [`manager.updatePreview()`](https://github.com/codesandbox/codesandbox-client/blob/master/standalone-packages/sandpack/src/manager/index.ts#L195) is called for a specific manager, it triggers an update to all the managers on the page.

### Demo

Live demo: https://two-managers-smooshpack.vercel.app/

Code: https://github.com/hzhu/two-managers-smooshpack/

_Notice:_ I expected one of the preview managers (foo preview) to update, but both preview managers are updated:

![2020-08-02 20 02 05](https://user-images.githubusercontent.com/1811365/89142217-a8acd400-d4fb-11ea-8bbc-bde0f338b4ac.gif)

### Discussion

Could someone explain the internal workings of why calling [`.updatePreview()` ](https://github.com/codesandbox/codesandbox-client/blob/master/standalone-packages/sandpack/src/manager/index.ts#L195)updates all of the managers on the page? Is there a way to render multiple preview managers on a single page and update them independently?

**Comment by CompuIves** [comment, marked as answer]
Oh yep, this is a bug. We should add an ID to the message so previews know what's intended for them and not for others. I can look at that today.

**Reply by hzhu** [reply]
@CompuIves, thank you for taking a look!

**Reply by CompuIves** [reply]
It's live now under version `0.0.66`!

**Reply by hzhu** [reply]
Woot woot that's incredible! Thanks @CompuIves! I'll give it a try this afternoon 🌞.

Edit: it works great!
